hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -307,7 +310,7 @@ public String toString() {
 	 * @param serviceRegistry The service registry
 	 */
 	public void contributeTypes(TypeContributions typeContributions, ServiceRegistry serviceRegistry) {
-		// by default, nothing to do
+		resolveLegacyLimitHandlerBehavior( serviceRegistry );
```

Comment:
Shouldn't we have the resolveLegacyLimitHandlerBehavior method called by `MetadataBuildingProcess` just like it does for `contributeTypes`? It's a little bit hacky that we resolve the limit handler during type contributions.